docs/using-modules.rst

```diff
@@ -0,0 +1,192 @@
+.. _modules:
+
+Modules
+#######
+
+A module is a set of function definitions and variable declarations which enables code reuse. Vyper favors code reuse through composition, rather than inheritance. A module encapsulates everything needed for code reuse, from type and function declarations to state.
```

Comment:
```suggestion
A module is a set of function definitions and variable declarations which enables code reuse. Vyper favors code reuse through composition, rather than inheritance. A module encapsulates everything needed for code reuse, from type and function declarations to state. It is important to note that functions which make use of defined state must be initialized in order to inherit that state, whereas functions that are "pure" do not require this.
```

Reply:
No resolution noted

Reply:
no, i added it, just updated the wording a bit https://github.com/vyperlang/vyper/pull/3947/commits/867098ac0a4b87dc5e91c26e23270142e0e69cfd